synapse/storage/databases/main/events_worker.py

```diff
@@ -1947,7 +1946,7 @@ def is_event_next_to_gap_txn(txn: LoggingTransaction) -> bool:
             # latest messages in the room so asking other homeservers for more
             # is useless. The new latest messages will just be federated as
             # usual.
-            txn.execute(forward_extremity_query, (event.room_id, event.event_id))
```

Comment:
Again, this points to brokenness if someone rolls forward and back again. This is only used by the experimental MSC3030 support, so this is *definitely* an edge case.